adafruit_rfm9x.py

```diff
@@ -576,32 +591,34 @@ def receive(self, timeout_s=0.5, keep_listening=True):
         # enough, however it's the best that can be done from Python without
         # interrupt supports.
         start = time.monotonic()
-        while not self.rx_done:
+        timed_out = False
+        while not timed_out and not self.rx_done:
             if (time.monotonic() - start) >= timeout_s:
-                return None  # Exceeded timeout.
-        # Clear interrupt.
-        self._write_u8(_RH_RF95_REG_12_IRQ_FLAGS, 0xFF)
+                timed_out = True
```

Comment:
Exception here too and it'll "return" early so you don't need the extra state.

Reply:
As above, there is still code that needs to execute after the timeout. 
More importantly, I don't think we want an exception here. The typical use of receive it to see if any packets are available. A timeout is normal here.  

We could break this into two parts like it is for Arduino and have an "available" function that the user polls until a packet is received.

Reply:
Ok, returning None is fine with me. I believe we do want an exception for `send` failure above though.